convex/lib/dynamicModelSwitching.ts

```
@@ -0,0 +1,337 @@
+ */
+export function analyzeComplexity(
+  message: string,
+  conversationHistory: Array<{ role: string; content: string }> = []
+): ComplexitySignals {
+  const lowerMessage = message.toLowerCase();
+
+  // Message characteristics
+  const messageLength = message.length;
+  const hasCodeBlocks = message.includes( "```" ) || message.includes( "`" );
+  const hasMultipleQuestions = ( message.match( /\?/g ) || [] ).length > 1;
+  const hasMath = /\d+[\+\-\*\/]\d+|equation|formula|calculate/.test( lowerMessage );
```

ESLint

[error] Unnecessary escape character: \+.

(no-useless-escape)